modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -372,7 +372,9 @@ public String extractIndexableAttributes(
 
 				Serializable value = field.getValue(locale);
 
-				if (Validator.isNull(value)) {
+				if ((value == null) ||
+					Objects.equals(value, StringPool.BLANK)) {
+
```

Comment:
@ericyanLr @noornajj why did you have to change from 
```
Validator.isNull(value)
``` 
to 
```
(value == null) || Objects.equals(value, StringPool.BLANK)
``` 

if `Validator.isNull(value)` returns `true` if value is `null` or `blank`? see [here](https://github.com/liferay/liferay-portal/blob/faf682e989d8872bdea67d3b17c2d4318c888502/portal-kernel/src/com/liferay/portal/kernel/util/Validator.java#L915-L919)

Reply:
This is to account for "null" literals. If `Validator.isNull(value)` is used, then if the field's value is "null" it'll be skipped

Reply:
@carolmariaabb, adding on to Noor's comment, do you know if "null" literals should be skipped or retained? This was a topic we weren't exactly sure about.

Reply:
@ericyanLr @noornajj In my opinion, we can skip "null" literals, because it seems like an edge case to me.. But you can ask in the #help-forms-data-engine slack channel to confirm.

Reply:
@carolmariaabb @ericyanLr I have posted the question in the channel and waiting for a reply